glean_parser/lint.py

```diff
@@ -543,17 +543,21 @@ def glinter(
     if parser_config is None:
         parser_config = {}
 
-    if lint_yaml_files(input_filepaths, file=file, parser_config=parser_config):
-        return 1
+    errors = 0
 
-    objs = parser.parse_objects(input_filepaths, parser_config)
+    nits = lint_yaml_files(input_filepaths, file=file, parser_config=parser_config)
+    errors += len(nits)
 
-    if util.report_validation_errors(objs):
-        return 1
+    objs = parser.parse_objects(input_filepaths, parser_config)
+    errors += util.report_validation_errors(objs)
 
     nits = lint_metrics(objs.value, parser_config=parser_config, file=file)
-    if any(nit.check_type == CheckType.error for nit in nits):
-        return 1
-    if len(nits) == 0:
+    errors += len(list(filter(lambda nit: nit.check_type == CheckType.error, nits)))
```

Comment:
This is a little more "modern" Python:

```suggestion
    errors += len([nit for nit in nits if nit.check_type == CheckType.error])
```